src/west/main.py

```diff
@@ -462,8 +469,16 @@ def get_external_commands():
                         format(spec.project.name, spec.name),
                         'this is a built in command')
                 continue
+            if spec.name in extension_commands:
+                log.wrn('ignoring project {} external command "{}";'.
+                        format(spec.project.name, spec.name),
+                        'command "{}" already defined as extension command'.
+                        format(spec.name))
+                continue
             filtered.append(spec)
+            extension_commands.add(spec.name)
         externals[path] = filtered
+        filtered = []
```

Comment:
How about just moving `filtered = []` to the line above `for spec in specs`, and deleting the line above `extension_commands = set()`? That might be a little easier to follow.